consensus/state.go

```diff
@@ -71,23 +92,31 @@ func CreateGenesisState(genesisTime Timestamp) (s *State) {
 	// Fill out the consensus informaiton for the genesis block.
 	s.currentBlockID = genesisBlock.ID()
 	s.currentPath[BlockHeight(0)] = genesisBlock.ID()
-	s.unspentOutputs[genesisBlock.MinerPayoutID(0)] = SiacoinOutput{
-		Value:     CalculateCoinbase(0),
-		SpendHash: ZeroAddress,
+	s.siacoinOutputs[genesisBlock.MinerPayoutID(0)] = SiacoinOutput{
+		Value:      CalculateCoinbase(0),
+		UnlockHash: ZeroAddress, // TODO: change to Nebulous Genesis Siacoin SpendHash Address
 	}
-	s.unspentSiafundOutputs[OutputID{0}] = SiafundOutput{
-		Value:            NewCurrency64(SiafundCount),
-		SpendHash:        ZeroAddress, // TODO: change to Nebulous Genesis Address
-		ClaimDestination: ZeroAddress, // TODO: change to Nebulous Genesis Address
+	s.siafundOutputs[SiafundOutputID{0}] = SiafundOutput{
+		Value:           NewCurrency64(SiafundCount),
+		UnlockHash:      ZeroAddress, // TODO: change to Nebulous Genesis Siafund SpendHash Address
+		ClaimUnlockHash: ZeroAddress, // TODO: change to Nebulous Genesis ClaimDestination Address
 	}
 
 	return
 }
 
+// RLock will readlock the state.
+//
+// TODO: Add a safety timer which will auto-unlock if the readlock is held for
+// more than a second. (panic in debug mode)
```

Comment:
1. function calls RLock, and RLock writes to a semaphore or something that tracks all of the locks in place
2. 1 second passes, RUnlock is automatically called and the semaphore is updated to show that the lock is gone.
3. function calls RUnlock, which checks the semaphore and realizes that someone screwed up. It doesn't call unlock and avoids the panic